src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -91,13 +91,44 @@ public static Checksum Create(ArrayBuilder<Checksum> checksums)
                 checksum.WriteTo(writer);
         });
 
+    public static Checksum CreateNew(ArrayBuilder<Checksum> checksums)
+    {
+        var checksumsCount = checksums.Count;
+        if (checksumsCount <= 100)
```

Comment:
@stephentoub does runtime have a general rule of thumb about reasonable stackalloc sizes?  for context, each item in this array is 16 bytes, so this would be up to 1.6k alloc (on a leaf node of the stack, no recursion).  Is that in line with what you guys do?

Reply:
We generally avoid going above 1k